libs/SalesforceSDK/src/com/salesforce/androidsdk/auth/idp/IDPAccountPickerActivity.java

```diff
@@ -0,0 +1,161 @@
+ * - Neither the name of salesforce.com, inc. nor the names of its contributors
+ * may be used to endorse or promote products derived from this software without
+ * specific prior written permission of salesforce.com, inc.
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.salesforce.androidsdk.auth.idp;
+
+import android.accounts.AccountManager;
+import android.app.Activity;
+import android.content.Intent;
+import android.os.Bundle;
+import android.text.TextUtils;
+import android.widget.Button;
+
+import com.salesforce.androidsdk.R;
+import com.salesforce.androidsdk.accounts.UserAccount;
+import com.salesforce.androidsdk.app.SalesforceSDKManager;
+import com.salesforce.androidsdk.ui.AccountSwitcherActivity;
+import com.salesforce.androidsdk.util.SalesforceSDKLogger;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * This class provides UI to select an existing signed in user account, or add a new account.
+ * It kicks off the IDP authentication flow once the selection is made. This screen is popped
+ * off the activity stack once the flow is complete, passing back control to the SP app.
+ *
+ * @author bhariharan
+ */
+public class IDPAccountPickerActivity extends AccountSwitcherActivity {
+
+    private static final int IDP_LOGIN_REQUEST_CODE = 999;
+    private static final String TAG = "IDPAccountPickerActivity";
+
+    private SPConfig spConfig;
+
+    @Override
+    public void onCreate(Bundle savedInstance) {
+        super.onCreate(savedInstance);
+        final Button button = findViewById(R.id.sf__switcher_apply_button);
+        button.setText(R.string.sf__idp_account_selector);
+
+        // Fetches the required extras.
+        final Intent intent = getIntent();
+        final Bundle extras = intent.getExtras();
+        if (extras != null) {
+            spConfig = new SPConfig(extras.getBundle(IDPCodeGeneratorActivity.SP_CONFIG_BUNDLE_KEY));
+        }
+    }
+
+    @Override
+    protected List<UserAccount> getAccounts() {
+        final List<UserAccount> accounts = userAccMgr.getAuthenticatedUsers();
+
+        // If no login server is passed in, return all user accounts.
+        if (TextUtils.isEmpty(spConfig.getLoginUrl())) {
```

Comment:
This is the part that does filtering based on login host (if a login host is passed in from the SP app).